clients/src/main/java/org/apache/kafka/common/protocol/types/Struct.java

```diff
@@ -28,6 +28,7 @@
 public class Struct {
     private final Schema schema;
     private final Object[] values;
+    private volatile ByteBuffer serialized = null;
```

Comment:
Please see my comment below. To add more to that comment, I am in favor of putting the cached ByteBuffer in `UpdateMetadataRequest` only if we override `toSend` there. There are two benefits:
1. Keep the hotfix changes in one place and simplify this patch.
2. We no longer need to cache the `Struct` because we will only call `toStruct` once and cache the serialized ByteBuffer for any future calls to `toSend`.